.github/workflows/build-test.yml

```
@@ -56,54 +56,49 @@ jobs:
         run: echo "BRANCH=${GITHUB_REF##*/}" >> $GITHUB_ENV
 
       - name: Build and save image tag
-        run: echo "IMAGE_TAG=${BRANCH}_terraform-${{ matrix.tf_version }}_awscli-${{ matrix.awscli_version }}" >> $GITHUB_ENV
+        run: echo "IMAGE_TAG=${BRANCH}_terraform-${{ matrix.tf_versions }}_awscli-${{ matrix.awscli_versions }}" >> $GITHUB_ENV
 
-      - name: Build image
-        run: docker image build . --file Dockerfile --build-arg TERRAFORM_VERSION=${{ matrix.tf_version }} --build-arg AWS_CLI_VERSION=${{ matrix.awscli_version }} --tag ${IMAGE_NAME}:${IMAGE_TAG}
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
 
-      - name: Save image
-        run: docker image save --output ${IMAGE_NAME}_${IMAGE_TAG}.tar ${IMAGE_NAME}:${IMAGE_TAG}
+      - name: Set up Buildx
+        uses: docker/setup-buildx-action@v2
 
-      - name: Upload image artifact
-        uses: actions/upload-artifact@v2
+      - name: Build and export container image
+        uses: docker/build-push-action@v3
         with:
-          name: ${{ env.IMAGE_NAME }}_${{ env.IMAGE_TAG }}
-          path: ${{ env.IMAGE_NAME }}_${{ env.IMAGE_TAG }}.tar
-
-  test:
-    runs-on: ubuntu-20.04
-    needs:
-      - build
-      - load_supported_versions
-    strategy:
-      matrix: ${{ fromJSON(needs.load_supported_versions.outputs.matrix) }}
-
-    steps:
-      - name: Checkout source
-        uses: actions/checkout@v3
-
-      - name: Save branch name as env var
-        run: echo "BRANCH=${GITHUB_REF##*/}" >> $GITHUB_ENV
-
-      - name: Build and save image tag
-        run: echo "IMAGE_TAG=${BRANCH}_terraform-${{ matrix.tf_version }}_awscli-${{ matrix.awscli_version }}" >> $GITHUB_ENV
-
-      - name: Download image artifact
-        uses: actions/download-artifact@v2
-        with:
-          name: ${{ env.IMAGE_NAME }}_${{ env.IMAGE_TAG }}
-
-      - name: Load image
-        run: docker image load --input ${{ env.IMAGE_NAME }}_${{ env.IMAGE_TAG }}.tar
+          context: .
+          platforms: linux/amd64
+          build-args: |
+            TERRAFORM_VERSION=${{ matrix.tf_versions }}
+            AWS_CLI_VERSION=${{ matrix.awscli_versions }}
+          tags: ${{ env.IMAGE_NAME }}:${{ env.IMAGE_TAG }}
+          push: false
+          load: true
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
 
       - name: Generate test config
         run: |
-          export TF_VERSION=${{ matrix.tf_version }}
-          export AWS_VERSION=${{ matrix.awscli_version }}
+          export tf_versions=${{ matrix.tf_versions }}
```

```suggestion
          export TF_VERSION=${{ matrix.tf_versions }}
```

Je ne suis pas sûr de bien comprendre, mais à mon avis ici, on serait bien sur la valorisation de la variable d'environnement vu qu'elle est utilisée en dessous par le `envsubst`, non ?

---

Houuula oui, bien vu 🙏  je corrige ça 👌 

Un grand merci pour ta revue 🤩